velox/vector/tests/VectorTestUtils.h

```diff
@@ -133,11 +134,35 @@ VectorGeneratedData<T> genTestData(
   testData.reserve(cardinality);
 
   for (int64_t m = 0; m < cardinality; m++) {
-    testData.push_back(VectorValueGenerator::cardValueOf<T>(
-        useFullTypeRange,
-        rng,
-        testData.stringViewBufferHolder(),
-        fixedWidthStringSize));
+    T value;
+    if constexpr (std::is_same_v<T, UnscaledShortDecimal>) {
```

Comment:
`cardValueOf` can be extended to support `UnscaledShortDecimal` `UnscaledLongDecimal` and we can remove this specialization.
`getDecimalPrecisionScale(type)` will work.

Reply:
Done.